src/map/status.cpp

```diff
@@ -12100,43 +12100,45 @@ int32 status_change_start(struct block_list* src, struct block_list* bl,enum sc_
 			val2 = 15 + 5 * val1; // AGI
 			val3 = 25; // Move speed increase
 			if (sd && (sd->class_&MAPID_BASEMASK) == MAPID_SUMMONER)
-				val4 = 10; // Ranged ATK increase
+				val4 = 10; // Ranged ATK increase if the target is a Doram
 			break;
 		case SC_SHRIMP:
 			val2 = 10; // BATK%, MATK%
 			break;
 		case SC_FRESHSHRIMP: {
 				int32 min = 0, max = 0;
+				map_session_data* ssd = reinterpret_cast<map_session_data*>(src);
```

Comment:
```suggestion
				map_session_data* ssd = BL_CAST( BL_PC, src );
```
Here you cannot use reinterpret_cast, since you cannot be sure that src is always a player (unless I missed something above).

Reply:
Correct !